lib/logstash/outputs/google_bigquery.rb

```diff
@@ -411,9 +424,13 @@ def initialize_uploader
               @logger.debug("BQ: reenqueue as log file is being currently appended to.",
                             :filename => filename)
               @upload_queue << filename
+
               # If we got here, it means that older files were uploaded, so let's
-              # wait another minute before checking on this file again.
-              sleep @uploader_interval_secs
+              # wait before checking on this file again.
+              #
+              # Use the min so we don't wait too long if the logs start rotating too quickly
+              # due to an increased number of events.
+              sleep [60, @uploader_interval_secs].min
```

Comment:
not sure why you cap it at 60 here?

Reply:
Based on a lot of the backlog issues around race conditions I didn't fully trust that there weren't other files in the queue if the head was the current file so I capped it down to the interval that the original comment said.

It's weird that the upload queue is used both for message passing between threads and as a staging area for data isn't meant to be processed yet.